src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  static FULL_HEALTH = 100;
+
+  constructor(name) {
+    this.name = name;
+
+    this.health = Animal.FULL_HEALTH;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  static HIDDEN = false;
+
+  constructor(name) {
+    super(name);
+
+    this.health = Herbivore.FULL_HEALTH;
+    this.hidden = Herbivore.HIDDEN;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.health = Carnivore.FULL_HEALTH;
```

Comment:
you don't need `this.health = Carnivore.FULL_HEALTH;` because you do this actions in parent class

Reply:
Delete this.health = Carnivore.FULL_HEALTH;